java/src/main/java/ai/rapids/cudf/JCudfSerialization.java

```diff
@@ -1235,8 +1282,31 @@ public static void writeToStream(ColumnVector[] columns, OutputStream out, long
                                    long numRows) throws IOException {
 
     ColumnBufferProvider[] providers = providersFrom(columns);
-    DataWriter writer = writerFrom(out);
-    writeSliced(providers, writer, rowOffset, numRows);
+    try {
+      DataWriter writer = writerFrom(out);
+      writeSliced(providers, writer, rowOffset, numRows);
+    } finally {
+      closeAll(providers);
+    }
+  }
+
+  /**
+   * Write all or part of a set of columns out in an internal format.
+   * @param columns the columns to be written.
+   * @param out the stream to write the serialized table out to.
+   * @param rowOffset the first row to write out.
+   * @param numRows the number of rows to write out.
+   */
+  public static void writeToStream(HostColumnVector[] columns, OutputStream out, long rowOffset,
+                                   long numRows) throws IOException {
+
+    ColumnBufferProvider[] providers = providersFrom(columns, false);
+    try {
+      DataWriter writer = writerFrom(out);
+      writeSliced(providers, writer, rowOffset, numRows);
+    } finally {
+      closeAll(providers);
```

Comment:
Similarly odd setup of making sure close methods, setup in this same method to do nothing, get called.